game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -1141,51 +1135,33 @@ private void updateUnitsThatCanMoveOnRoute(final Collection<Unit> units, final R
     List<Unit> best = new ArrayList<>(units);
     // if the player selects a land unit and other units when the
     // only consider the non land units
-    if (route.getStart().isWater()
-        && route.getEnd() != null
```

Comment:
Side-note: if we have proved that  `route.getEnd() != null` is always false, it might be good to do a global cleanup of this check in one pass everywhere.

Reply:
Sure, but right now my goal is to land the move lag fix - and this was a refactoring I've split off from that change.